### PR TITLE
[ANCHOR-397] Add missing claims to SEP-24 JWTs

### PR DESCRIPTION
**Pull request by philipliu**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

This sets `client_name` and `client_domain` for the JWTs generated by SEP-24. It also adds the missing `sub` claim to the more info URL JWT.

### Why

These claims are used to identify the wallet and user.

### Known limitations

N/A

### CODE REVIEW

File: platform/src/main/java/org/stellar/anchor/platform/service/SimpleInteractiveUrlConstructor.java [hunk 1]
```diff
@@ -70,12 +75,15 @@ String constructToken(Sep24Transaction txn, Map<String, String> request) {
         (isEmpty(txn.getSep10AccountMemo()))
             ? txn.getSep10Account()
             : txn.getSep10Account() + ":" + txn.getSep10AccountMemo();
+    ClientsConfig.ClientConfig clientConfig =
```
Comment (Ifropc) [inline platform/src/main/java/org/stellar/anchor/platform/service/SimpleInteractiveUrlConstructor.java:78]: For custodial wallets you should also get name by account (transaction may not have a clientDomain)
If clientDomain is missing, should validate that client type is custodial

File: platform/src/main/java/org/stellar/anchor/platform/service/SimpleMoreInfoUrlConstructor.java [hunk 2]
```diff
@@ -1,35 +1,48 @@
...
 
   @Override
   @SneakyThrows
-  public String construct(Sep24Transaction txn) throws URISyntaxException, MalformedURLException {
+  public String construct(Sep24Transaction txn) {
+    String account =
+        isEmpty(txn.getSep10AccountMemo())
+            ? txn.getSep10Account()
+            : txn.getSep10Account() + ":" + txn.getSep10AccountMemo();
+    ClientsConfig.ClientConfig clientConfig =
```
Comment (Ifropc) [inline platform/src/main/java/org/stellar/anchor/platform/service/SimpleMoreInfoUrlConstructor.java:37]: Same here

File: platform/src/test/kotlin/org/stellar/anchor/platform/service/SimpleInteractiveUrlConstructorTest.kt [hunk 3]
```diff
@@ -59,17 +69,25 @@ class SimpleInteractiveUrlConstructorTest {
...
+    assertEquals("lobstr", claims["client_name"] as String)
 
     testTxn.sep10AccountMemo = null
+    testTxn.clientDomain = "unknown.com"
     jwt = parseJwtFromUrl(constructor.construct(testTxn, testRequest))
+    claims = jwt.claims
     testJwt(jwt)
     assertEquals("GBLGJA4TUN5XOGTV6WO2BWYUI2OZR5GYQ5PDPCRMQ5XEPJOYWB2X4CJO", jwt.sub)
+    assertEquals("unknown.com", claims["client_domain"] as String)
+    assertEquals("", claims["client_name"] as String)
```
Comment (Ifropc) [inline platform/src/test/kotlin/org/stellar/anchor/platform/service/SimpleInteractiveUrlConstructorTest.kt:90]: Can you add test for custodial wallet? It won't have a client_domain in the challenge transaction

**Review by Ifropc — CHANGES_REQUESTED** [review]

File: platform/src/main/java/org/stellar/anchor/platform/service/SimpleInteractiveUrlConstructor.java (hunk 1 above)
Comment (philipliu) [reply platform/src/main/java/org/stellar/anchor/platform/service/SimpleInteractiveUrlConstructor.java:78]: By account, do you mean the signing key in the client config?

Comment (Ifropc) [reply platform/src/main/java/org/stellar/anchor/platform/service/SimpleInteractiveUrlConstructor.java:78]: Yes

Comment (philipliu) [reply platform/src/main/java/org/stellar/anchor/platform/service/SimpleInteractiveUrlConstructor.java:73]: Done

File: platform/src/main/java/org/stellar/anchor/platform/service/SimpleMoreInfoUrlConstructor.java (hunk 2 above)
Comment (philipliu) [reply platform/src/main/java/org/stellar/anchor/platform/service/SimpleMoreInfoUrlConstructor.java:32]: Done

File: platform/src/test/kotlin/org/stellar/anchor/platform/service/SimpleInteractiveUrlConstructorTest.kt (hunk 3 above)
Comment (philipliu) [reply platform/src/test/kotlin/org/stellar/anchor/platform/service/SimpleInteractiveUrlConstructorTest.kt]: Done

**Review by Ifropc — APPROVED** [review]

File: platform/src/test/kotlin/org/stellar/anchor/platform/config/Sep10ConfigTest.kt [hunk 4]
```diff
@@ -201,52 +224,4 @@ class Sep10ConfigTest {
     config.postConstruct()
     assertEquals("localhost:8080", config.webAuthDomain)
   }
-
-  companion object {
-    @JvmStatic
-    fun validOmnibusAccounts(): Stream<List<String>> {
```
Comment (lijamie98) [inline platform/src/test/kotlin/org/stellar/anchor/platform/config/Sep10ConfigTest.kt:207]: Are these functions not used?

**Review by lijamie98 — APPROVED** [review]
LGTM with questions about the removal of `validOmnibusAccounts` and `invalidOmnibusAccounts` functions.

File: platform/src/test/kotlin/org/stellar/anchor/platform/config/Sep10ConfigTest.kt (hunk 4 above)
Comment (philipliu) [reply platform/src/test/kotlin/org/stellar/anchor/platform/config/Sep10ConfigTest.kt:207]: Nope, I think it was left there unintentionally. We can add it back if we need it in the future.

**Pull request merged**